tests/integration/synapseclient/core/test_external_storage.py

```diff
@@ -0,0 +1,190 @@
+import importlib
+import os
+import tempfile
+import uuid
+
+from synapseclient import File, Folder
+from synapseclient.core.retry import with_retry
+
+from nose.tools import assert_equal, assert_raises, assert_true
+import unittest
+from tests import integration
+
+try:
+    boto3 = importlib.import_module('boto3')
+except ImportError:
+    boto3 = None
+
+
+def check_test_preconditions():
+    # in order ot run the tests in this file boto3 must be available
+    # and we must have configuration indicating where to store external
+    # storage for testing
+
+    skip_tests = False
+    reason = ''
+    if not boto3:
+        skip_tests = True
+        reason = 'boto3 not available'
+
+    elif not (
+        os.environ.get('EXTERNAL_S3_BUCKET_NAME') and
+        os.environ.get('EXTERNAL_S3_BUCKET_AWS_ACCESS_KEY_ID') and
+        os.environ.get('EXTERNAL_S3_BUCKET_AWS_SECRET_ACCESS_KEY')
+    ):
+        skip_tests = True
+        reason = 'External bucket access not defined in environment'
+
+    return skip_tests, reason
+
+
+def setup(module):
+    module.syn = integration.syn
+    module.project = integration.project
+
+
+def get_aws_env():
+    return os.environ['EXTERNAL_S3_BUCKET_NAME'], {
+        'aws_access_key_id': os.environ['EXTERNAL_S3_BUCKET_AWS_ACCESS_KEY_ID'],
+        'aws_secret_access_key': os.environ['EXTERNAL_S3_BUCKET_AWS_SECRET_ACCESS_KEY'],
+    }
+
+
+@unittest.skipIf(*check_test_preconditions())
+class ExernalStorageTest(unittest.TestCase):
+
+    @classmethod
+    def _make_temp_file(cls, contents=None, **kwargs):
+        tmp_file = tempfile.NamedTemporaryFile(**kwargs)
+
+        if contents:
+            with open(tmp_file.name, 'w') as f:
+                f.write(contents)
+
+        return tmp_file
+
+    def _prepare_bucket_location(self, key_prefix):
+        bucket_name, aws_creds = get_aws_env()
+        s3_client = boto3.client('s3', **aws_creds)
+        s3_client.put_object(
+            Body=syn.credentials.username,
+            Bucket=bucket_name,
+            Key=f"{key_prefix}/owner.txt"
+        )
+
+        return s3_client
+
+    def _configure_storage_location(self, sts_enabled=False):
+        folder_name = str(uuid.uuid4())
+        s3_client = self._prepare_bucket_location(folder_name)
+
+        folder = syn.store(Folder(name=folder_name, parent=integration.project))
```

Comment:
Create a convenience function for creating storage location and enabling it for STS?